src/check_datapackage/custom_check.py

```diff
@@ -142,7 +199,9 @@ def apply_extensions(
     Returns:
         A list of `Issue`s.
     """
+    extensions_as_one = chain(extensions.required_checks, extensions.custom_checks)
     return _flat_map(
-        extensions,
-        lambda extension: extension.apply(properties),
+        extensions_as_one,
+        # TODO: fix type ignore when we use Protocols/ABCs
+        lambda extension: extension.apply(properties),  # type: ignore[attr-defined]
     )
```

Comment:
```suggestion
    all_extensions: list[RequiredCheck | CustomCheck] = (
        extensions.required_checks + extensions.custom_checks
    )
    return _flat_map(
        all_extensions,
        lambda extension: extension.apply(properties),
    )
```

If you declare the type explicitly, it will know that `apply` exists even without a Protocol or super class

Reply:
Ah, for some reason I thought I tried that and it didn't register. But I must have used `chain()` not `+`. Seems to work now! :+1: